packages/concerto-core/lib/introspect/decorated.js

```diff
@@ -38,13 +38,14 @@ class Decorated {
      * Create a Decorated from an Abstract Syntax Tree. The AST is the
      * result of parsing.
      *
-     * @param {ModelFile} modelFile - the model file
+     * @param {ModelFile|undefined} modelFile - the model file
```

Comment:
Do we really want `|undefined` here?

Reply:
I was hoping that `ModelFile` could be a `Decorated` as per other decorated objects, however in the constructor for `ModelFile` you cannot pass `this` into the `super(modelFile, ast)`. 

The original approach was if you do not pass a `ModelFile` into the constructor (passing `undefined`), you can pass it later via `Decorated.setModelFile`, and if you've done neither of those things `Decorated.process` will throw an error.

I think this is actually cleaner if `Decorated.getModelFile` is abstract and must be implemented by the subclass, where `ModelFile.getModelFile` can return `this`. See new commit!